pilot/pkg/config/mcp/controller.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coredatamodel
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

Comment:
Please remove and find alternative for meta_v1.Time{}. We try to keep k8s contained in k8s package

Reply:
The [core Pilot data model](https://github.com/istio/istio/blob/6acd9fd9ed1fa3bf7f77940ed0a7a5ee24fe30a3/pilot/pkg/model/config.go#L41-L87) uses this k8s metadata type. It looks like that should be pretty easy to clean-up to use `time.Time` directly.

Reply:
Can we make a separate PR for this?

Reply:
https://github.com/istio/istio/pull/8195

Reply:
Thanks, I will wait for the PR to go through to rebase.